include/circt/Dialect/HW/HWOpInterfaces.h

```diff
@@ -77,25 +84,38 @@ class InnerRefNamespace : public TraitBase<ConcreteType, InnerRefNamespace> {
 template <typename ConcreteType>
 class InnerSymbolTable : public TraitBase<ConcreteType, InnerSymbolTable> {
 public:
-  static LogicalResult verifyRegionTrait(Operation *op) {
-    // Insist that ops with InnerSymbolTable's provide a Symbol, this is
-    // essential to how InnerRef's work.
-    static_assert(
-        ConcreteType::template hasTrait<::mlir::SymbolOpInterface::Trait>(),
-        "expected operation to define a Symbol");
-
-    // InnerSymbolTable's must be directly nested within an InnerRefNamespace.
-    auto *parent = op->getParentOp();
-    if (!parent || !isa<circt::hw::InnerRefNamespaceLike>(parent))
-      return op->emitError(
-          "InnerSymbolTable must have InnerRefNamespace parent");
-
-    return success();
-  }
+  static LogicalResult verifyRegionTrait(Operation *op);
 };
 } // namespace OpTrait
 } // namespace mlir
 
 #include "circt/Dialect/HW/HWOpInterfaces.h.inc"
 
+namespace mlir {
+namespace OpTrait {
+template <typename ConcreteType>
+LogicalResult InnerSymbolTable<ConcreteType>::verifyRegionTrait(Operation *op) {
```

Comment:
I don't think this code belongs under `verifyRegion`s.  This is usually used when the parent op needs to verify something about op's in its region.  I think this can be moved to the regular trait verifier.